core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/mercury.go

```diff
@@ -117,6 +110,69 @@ func (l *StreamsLookup) IsMercuryV03() bool {
 	return l.FeedParamKey == FeedIDs
 }
 
-func (l *StreamsLookup) IsMercuryUsingBatchPathV03() bool {
+func (l *StreamsLookup) IsMercuryV03UsingBlockNumber() bool {
 	return l.TimeParamKey == BlockNumber
 }
+
+type Packer interface {
+	UnpackCheckCallbackResult(callbackResp []byte) (uint8, bool, []byte, uint8, *big.Int, error)
+	PackGetUpkeepPrivilegeConfig(upkeepId *big.Int) ([]byte, error)
+	UnpackGetUpkeepPrivilegeConfig(resp []byte) ([]byte, error)
+	DecodeStreamsLookupRequest(data []byte) (*StreamsLookupError, error)
+}
```

Comment:
do we still need this interface? I think we could suffice with just the `abiPacker` struct below since this is all internal to the `mercury` package now.

Reply:
Yes, I think you are very right on this! the interface is not mandatory.

I am still learning golang, and what I am following here is to have interface defining the functions (what), while structs focus on implementation (how). Now we have the Packer interface defining `what` and the abiPacker struct defining `how`. When using it in streams struct, we use the interface `packer          mercury.Packer`, which is the `what`. 

Please correct me.

Reply:
Sync-ed offline, the packer is also used in the streams package. Better to keep it the way it is now.